### PR TITLE
expose properties on query spec object and interface out smartstore s…

### PR DESCRIPTION
**Pull request by jhorstsf**
…o it could be mocked by consumers

### CODE REVIEW

**Comment by jhorstsf** [comment]
@kchitalia

File: SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs [hunk 1]
```diff
@@ -165,6 +166,16 @@ public object Upsert(string soupName, string soupElt, string externalIdPath, boo
             return NativeSmartStore.Upsert(soupName, JObject.Parse(soupElt), externalIdPath, handleTx);
         }
 
+        public bool BeginDatabaseTransaction()
```
Comment (kchitalia) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs]: You can get to this method via methods in Native smartstore, do we really need to expose this?

File: SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs [hunk 2]
```diff
@@ -165,6 +166,16 @@ public object Upsert(string soupName, string soupElt, string externalIdPath, boo
...
+        {
+            return NativeSmartStore.Database.BeginTransaction();
+        }
+
+        public bool CommitDatabaseTransaction()
```
Comment (kchitalia) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs]: Same for this method.

File: SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs (hunk 1 above)
Comment (jhorstsf) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs]: discussed offline with @kchitalia, yes we need this to perfom bulk operations efficiently

**Comment by kchitalia** [comment]
:+1:

**Pull request merged**